internal/v1/handler_blueprints.go

```diff
@@ -405,13 +405,28 @@ func (h *Handlers) GetBlueprint(ctx echo.Context, id openapi_types.UUID, params
 	return ctx.JSON(http.StatusOK, blueprintResponse)
 }
 
-func (h *Handlers) lintBlueprint(ctx echo.Context, blueprint *BlueprintBody, fixup bool) ([]BlueprintLintItem, error) {
+func (h *Handlers) lintBlueprint(ctx echo.Context, blueprintEntry *db.BlueprintEntry, fixup bool) ([]BlueprintLintItem, error) {
 	lintErrors := []BlueprintLintItem{}
-	if blueprint.Customizations.Openscap != nil {
+	bpBody, err := BlueprintFromEntry(blueprintEntry)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
2/2 And here, we are calling it again. I kinda do not like this, BlueprintFromEntry calls json.Unmarshal. I think I would even prefer passing both BlueprintEntry and BlueprintBody to the function rather than calling it twice, but that's maybe a subject to more discussion. I think we might aim to figure this out so that we do not have to do this in the first place.

Reply:
@avitova You're absolutely right about the duplicate unmarshalling being inefficient. I appreciate your suggestion to pass blueprintEntry.ServiceSnapshots separately.
After considering both approaches, I decided to align with @lzap's suggestion to pass only `blueprintEntry` for these reasons:

1. Passing a single parameter that contains all needed data follows Go conventions and keeps the function signature simple.
2. while the duplicate unmarshalling does add overhead, the cost is relatively small compared to the compliance service calls. The cleaner API makes the code easier to understand and maintain long-term.